### PR TITLE
Extend and improve OCI terraform README

### PR DESCRIPTION
**Pull request by iok0**
Now includes a section on running the terraform via OCI cloud shell

### CODE REVIEW

**Comment by sthetland** [comment]
I made a pass at the formatting and light edit. Otherwise, looks good to me.

**Review by credibleforce — APPROVED** [review]
Changes look good to me @sthetland

File: lw-oci-integration/tf/README.md [hunk 1]
```diff
@@ -1,34 +1,103 @@
...
+4. OCI auth is automatically handled by the cloud shell. To use this, we modify a couple of files:
+   * In `main.tf`, edit the provider section so that it looks like this (setting `tenancy_ocid` and `region` accordingly):
+     
+     ```
+     provider "oci" {
+       tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaatxph3zpu3jkem4cseejktwllb6dgsam7cx3jy63pazejzobvjqqq"
+       region = "us-sanjose-1"
+     } 
+     
+     ```
```
Comment (credibleforce) [inline lw-oci-integration/tf/README.md:45]: Hmmm, actually the `main.tf` today uses the variables from tfvars to set the provider details:
```
provider "oci" {
  tenancy_ocid     = var.tenancy_ocid
  user_ocid        = var.user_ocid
  fingerprint      = var.fingerprint
  private_key_path = var.private_key_path
  region           = var.region
}
```

I don't think we need to or should recommend hard coding these values.

**Review by credibleforce — CHANGES_REQUESTED** [review]
I think we should remove the section about updating `main.tf` here in favour of using the tfvars for the provider details.

**Review by fdong-lw — COMMENTED** [review]
same comment to replace hard-code vars

File: lw-oci-integration/tf/README.md (hunk 1 above)
Comment (iok0) [reply lw-oci-integration/tf/README.md]: Thanks - I see main.tf change has been reverted and instead the instruction for cloudshell is to set non-required variables to null:
```
     user_ocid=null
     fingerprint=null
     private_key_path=null
```

**Review by sthetland — APPROVED** [review]

**Review by fdong-lw — APPROVED** [review]
LG

File: lw-oci-integration/tf/README.md [hunk 2]
```diff
@@ -1,34 +1,90 @@
...
-```
-6. After the plan is applied, a Lacework cloud account policy is created locally under ~/.oci/lacework_cloud_account.json. You can post this file to Lacework via the Lacework CLI to create your integration:
-
-```
- lacework api post /api/v2/CloudAccounts -d "$(cat ~/.oci/lacework_cloud_account.json)"
-```
-
-
-| Note          |
+You can run the Terraform scripts locally or via an OCI cloud shell. We recommend using the OCI cloud shell, since it comes with Terraform pre-installed and provides for authentication.
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:4]:
```suggestion
The OCI Terraform scripts can be run locally or via an OCI cloud shell. Using the OCI cloud shell is the recommended option, as it comes with Terraform pre-installed and aids in authentication.
```

File: lw-oci-integration/tf/README.md [hunk 3]
```diff
@@ -1,34 +1,90 @@
...
+
+## Prepare to run Terraform locally (Option 1)
+
+Before using the Terraform script, you must create the required environment variables to access your OCI tenant. The recommended way to do this is:
+
+1. Clone this github repo. For example:
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:10]: Should we have a pre-reqs section here to have git & Terraform installed locally?
Also what is the min version of TF that this has been tested with so far that we could call out as the min supported version?

File: lw-oci-integration/tf/README.md [hunk 4]
```diff
@@ -1,34 +1,90 @@
...
+2. Change directory to the `lw-oci-integration/tf` folder and copy the `vars.tfvars.example` to `terraform.tfvars`.
+3. Update the values in this file to align with your administrative OCI user access.
+    ```
+    region="<YOUR HOME REGION  (e.g. us-ashburn-1)>"
+    tenancy_ocid="<YOUR OCI TEANCY OCID>"
+    user_ocid="<YOUR OCI USER OCID>"
+    fingerprint="<YOUR KEY FINGER PRINT>"
+    private_key_path="<YOUR KEY KEY PATH (e.g. ~/.oci/myocikey.pem)>"
+    ```
+4. Install the Lacework CLI (if it is not already installed) and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:23]: "(if it is not already installed)" seems unnecessary IMO
```suggestion
4. Install and configure the Lacework CLI with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).
```

Comment (rmoles) [reply lw-oci-integration/tf/README.md:23]: Second suggestion
```suggestion
4. Install the Lacework CLI and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).
```

File: lw-oci-integration/tf/README.md [hunk 5]
```diff
@@ -1,34 +1,90 @@
...
+     
+     ```
+     region="<YOUR HOME REGION  (e.g. us-ashburn-1)>"
+     tenancy_ocid="<YOUR OCI TEANCY OCID>"
+     user_ocid=null
+     fingerprint=null
+     private_key_path=null
+     
+     ```
+5. Install the Lacework CLI (if it is not already installed) and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:49]: Same suggestion as above

File: lw-oci-integration/tf/README.md [hunk 6]
```diff
@@ -1,34 +1,90 @@
...
+
+1. Open the OCI cloud shell. For example:
+   1. Log in to the OCI console.
+   2. Click on the '<>' icon (developer tools), located on the top right of the OCI web console.
+   3. Select Cloud Shell.
+2. Clone this github repo. For example:
+    ```
+    git clone https://github.com/lacework-dev/scripts.git
+    ```
+3. Change directory to the `lw-oci-integration/tf` folder and copy `vars.tfvars.example` to `terraform.tfvars`.
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:35]:
```suggestion
3. Change directory to the `scripts/lw-oci-integration/tf` folder and copy `vars.tfvars.example` to `terraform.tfvars`.
    ```
    cd scripts/lw-oci-integration/tf
   cp  vars.tfvars.example terraform.tfvars
    ```
```

File: lw-oci-integration/tf/README.md [hunk 7]
```diff
@@ -1,34 +1,90 @@
...
+4. OCI auth is automatically handled by the cloud shell. To use this, we modify a couple of files:
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:36]:
```suggestion
4. OCI auth is automatically handled by the cloud shell. To use this, modify the following files:
```

Comment (rmoles) [reply lw-oci-integration/tf/README.md:36]: Actually, it looks like we only edit `terraform.tfvars` in which case my suggestion would be
```suggestion
4. OCI auth is automatically handled by the cloud shell. To use this, modify the values in `terraform.tfvars`:
```

File: lw-oci-integration/tf/README.md [hunk 8]
```diff
@@ -1,34 +1,90 @@
...
+   * In `terraform.tfvars`, set the `region` and `tenancy_ocid` as appropriate. The `user_ocid`, `fingerprint` 
+and `private_key_path` fields are not applicable when using cloud shell. The `group_name`, `user_name` and `policy_name` will be
+created by the Terraform script for the Lacework OCI integration. They can be left as they are, or modified if required.
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:39]:
```suggestion
   * `region` and `tenancy_ocid` should be set as appropriate.
   * `user_ocid`, `fingerprint` and `private_key_path` fields are not applicable when using cloud shell and should be set to null.
   * `group_name`, `user_name` and `policy_name` will be
created by the Terraform script for the Lacework OCI integration. They can be left as they are, or modified if required.
```

File: lw-oci-integration/tf/README.md [hunk 9]
```diff
@@ -1,34 +1,90 @@
...
+   2. Install to the present working directory (PWD):
+      ```
+      ./install.sh -d $PWD
+      ```
+   3. Modify the path (temporary) to include the PWD:
+      ```
+      export PATH=$PATH:$PWD
+      ```
+
+Now, continue following instructions on [Lacework CLI Guide](https://docs.lacework.net/cli/) to configure your API key.
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:65]: This is already linked in the text for step 5, should we remove this link?

File: lw-oci-integration/tf/README.md [hunk 10]
```diff
@@ -1,34 +1,90 @@
...
+
+# Running Terraform (common to options 1 and option 2)
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:67]: IMO it is implied that it is common
```suggestion
# Running Terraform
```

File: lw-oci-integration/tf/README.md [hunk 11]
```diff
@@ -1,34 +1,90 @@
...
+
+1. Once the file is updated, run the following command to initialize Terraform:
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:69]: Which file?

File: lw-oci-integration/tf/README.md [hunk 12]
```diff
@@ -1,34 +1,90 @@
...
+   ```
+   terraform init
+   ```
+
+2. Now verify and plan the Terraform using this command:
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:74]:
```suggestion
2. Now verify and generate a Terraform plan:
```

File: lw-oci-integration/tf/README.md [hunk 13]
```diff
@@ -1,34 +1,90 @@
...
+   ```
+   terraform plan
+   ```
+3. If everything looks good, apply the change using the following command:
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:78]: Again, this is slightly vague... Maybe something along the lines of:
```suggestion
3. If `Terraform Plan` runs successfully with no errors, lets run the Terraform to create the required OCI resources:
```

File: lw-oci-integration/tf/README.md [hunk 14]
```diff
@@ -1,34 +1,90 @@
...
+   ```
+   terraform apply -auto-approve
+   ```
+4. After the plan is applied, a Lacework cloud account policy is created locally under ~/.oci/lacework_cloud_account.json. You can post this file to Lacework via the Lacework CLI to create your integration:
```
Comment (rmoles) [inline lw-oci-integration/tf/README.md:82]:
```suggestion
4. After the the Terraform is applied, a Lacework cloud account payload is created locally under ~/.oci/lacework_cloud_account.json. In order to create your OCI integration on the Lacework platform, run the following post: 
```

File: lw-oci-integration/tf/README.md (hunk 2 above)
Comment (benradstone) [reply lw-oci-integration/tf/README.md:4]:
```suggestion
The OCI Terraform scripts can be run locally or through an OCI cloud shell. Using the OCI cloud shell is recommended, as it comes with Terraform pre-installed and helps with authentication.
```

File: lw-oci-integration/tf/README.md (hunk 4 above)
Comment (benradstone) [reply lw-oci-integration/tf/README.md:23]: Yes that's fine. Users will ignore the advice to install the CLI if they already have it. No need to call it explicitly.

File: lw-oci-integration/tf/README.md (hunk 7 above)
Comment (benradstone) [reply lw-oci-integration/tf/README.md:36]:
```suggestion
4. OCI authentication is automatically handled by the cloud shell. To use this, modify the values in `terraform.tfvars`:
```

File: lw-oci-integration/tf/README.md (hunk 8 above)
Comment (benradstone) [reply lw-oci-integration/tf/README.md:39]:
```suggestion
   * `region` and `tenancy_ocid` should be set as appropriate.
	   * `user_ocid`, `fingerprint` and `private_key_path` fields are not applicable when using cloud shell and should be set to null.
	   * `group_name`, `user_name` and `policy_name` will be created by the Terraform script for the Lacework OCI integration. They can be left as they are, or modified if required.
```

File: lw-oci-integration/tf/README.md (hunk 10 above)
Comment (benradstone) [reply lw-oci-integration/tf/README.md:67]: Yes that's usually better practice.

File: lw-oci-integration/tf/README.md (hunk 13 above)
Comment (benradstone) [reply lw-oci-integration/tf/README.md:78]:
```suggestion
3. If `Terraform Plan` runs successfully with no errors, use the following command to create the required OCI resources:
```

File: lw-oci-integration/tf/README.md (hunk 14 above)
Comment (benradstone) [reply lw-oci-integration/tf/README.md:82]:
```suggestion
4. After the Terraform is applied, a Lacework cloud account payload is created locally under `~/.oci/lacework_cloud_account.json`. To create your OCI integration on the Lacework platform, run the following post:
```

**Review by benradstone — COMMENTED** [review]
Left a few comments for suggested changes.

File: lw-oci-integration/tf/README.md (hunk 11 above)
Comment (iok0) [reply lw-oci-integration/tf/README.md:73]: Remove "Once the file is updated,"

**Pull request merged**